lib/index.js

```diff
@@ -190,8 +192,11 @@ exports.register = function (server, options, next) {
 
     server.ext('onPreResponse', function (request, reply) {
 
+        var skipCacheStorage = !settings.errorOnCacheNotReady && !cache.isReady();
+
         if (!request.session ||
-            (!request.session._isModified && !request.session._isLazy)) {
+            skipCacheStorage ||
```

Comment:
Done, the implementation is a bit wordy, so I'll refactor tomorrow when I pull out sinon.